src/server/scripts/Northrend/Naxxramas/boss_gluth.cpp

```diff
@@ -110,47 +133,333 @@ class boss_gluth : public CreatureScript
+                    if (damage > 0)
+                        GetOriginalCaster()->CastCustomSpell(SPELL_DECIMATE_DMG, SPELLVALUE_BASE_POINT0, damage, unit);
                 }
             }
+        }
+
+        // This method handles the change of the zombie's behavior after a decimate.
+        void HandleSendEvent(SpellEffIndex /* index */)
+        {
+            if (GetOriginalCaster()->GetEntry() == NPC_GLUTH)
+                GetOriginalCaster()->ToCreature()->GetAI()->DoAction(DO_ZOMBIE_RUN);
+        }
+
+        bool Validate(SpellInfo const* /*spellInfo*/) override
+        {
+            if (sSpellMgr->GetSpellInfo(SPELL_DECIMATE_DMG))
+                return true;
+            else
+                return false;
+        }
+
+        void Register() override
+        {
+            OnEffectHitTarget += SpellEffectFn(spell_gluth_decimate_SpellScript::HandleScriptEffect, EFFECT_0, SPELL_EFFECT_SCRIPT_EFFECT);
+            OnEffectHit += SpellEffectFn(spell_gluth_decimate_SpellScript::HandleSendEvent, EFFECT_2, SPELL_EFFECT_SEND_EVENT);
+        }
+    };
+
+    SpellScript* GetSpellScript() const
+    {
+        return new spell_gluth_decimate_SpellScript();
+    }
+
+};
+
+class spell_gluth_zombiechow_search : public SpellScriptLoader
+{
+public:
+    spell_gluth_zombiechow_search() : SpellScriptLoader("spell_gluth_zombiechow_search") { }
+
+    class spell_gluth_zombiechow_search_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_gluth_zombiechow_search_SpellScript);
+
+        void HealForEachTargetHit()
+        {
+            GetOriginalCaster()->ModifyHealth(int32(GetOriginalCaster()->CountPctFromMaxHealth(5)));
+        }
+        
+        bool Validate(SpellInfo const* /*spellInfo*/) override
+        {
+            if (sSpellMgr->GetSpellInfo(SPELL_ZOMBIE_CHOW_SEARCH_MULTI) && sSpellMgr->GetSpellInfo(SPELL_ZOMBIE_CHOW_SEARCH_SINGLE))
+                return true;
             else
+                return false;
+        }
+
+        void Register() override
+        {
+            AfterHit += SpellHitFn(spell_gluth_zombiechow_search_SpellScript::HealForEachTargetHit);
+        }
+    };
+
+    SpellScript* GetSpellScript() const
+    {
+        return new spell_gluth_zombiechow_search_SpellScript();
+    }
+
+};
+
+// creature 16360 (10man) / 30303 (25man)
+class npc_zombie_chow : public CreatureScript
+{
+public:
+
+    npc_zombie_chow() : CreatureScript("npc_zombie_chow") { }
+
+    struct npc_zombie_chowAI : public ScriptedAI
+    {
+        npc_zombie_chowAI(Creature* creature) : ScriptedAI(creature)
+        {
+            GluthGUID = creature->GetInstanceScript()->GetGuidData(DATA_GLUTH);
+
+            DoCast(me, SPELL_INFECTED_WOUND);
+            state = STATE_ZOMBIE_NORMAL;
+        }
+
+        void UpdateAI(uint32 /*diff*/) override
+        {
+            if (!UpdateVictim())
+                return;
+
+            Creature* gluth = ObjectAccessor::GetCreature(*me, GluthGUID);
```

Comment:
line 391. there is a need for a reference to gluth's Creature*. I guess I could move that line into the if at 387 though.